<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Function reference â€¢ metacell</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-index">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">metacell</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/a-basic_pbmc8k.html">Usage</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li class="dropdown-header">Core</li>
    <li>
      <a href="../articles/a-basic_pbmc8k.html">Running metacell analysis: guided tutorial on 8K PBMCs</a>
    </li>
    <li>
      <a href="../articles/b-geneset_by_anchor_pbmc8k.html">Supervised filtering of feature genes: guided tutorial on 8K PBMCs</a>
    </li>
    <li>
      <a href="../articles/c-metacell_annotation.html">Annotation of metacells</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Misc</li>
    <li>
      <a href="../articles/d-amphimedon.html">Analyzing whole-organism scRNA-seq data with metacell</a>
    </li>
    <li>
      <a href="../articles/e-data_convert.html">UMI matrix conversion to standard formats</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a></a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index"><colgroup><col class="alias"><col class="title"></colgroup><tbody><tr><th colspan="2">
          <h2 id="metacell-pipeline-functions">metacell pipeline functions <a href="#metacell-pipeline-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="mcell_add_cgraph_bknn_norm_mc.html">mcell_add_cgraph_bknn_norm_mc()</a></code> </p>
        </td>
        <td><p>Build a cell graph using balanced knn after normalizing over a given metacell model</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_add_cgraph_from_distmat.html">mcell_add_cgraph_from_distmat()</a></code> </p>
        </td>
        <td><p>Build a cell graph using blanacing of an extrenal distance matrix</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_add_cgraph_from_mat_bknn.html">mcell_add_cgraph_from_mat_bknn()</a></code> </p>
        </td>
        <td><p>Build a cell graph using balanced knn graph on given gene features</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_add_cgraph_from_mat_raw_knn.html">mcell_add_cgraph_from_mat_raw_knn()</a></code> </p>
        </td>
        <td><p>Build a cell graph using raw knn graph on given gene features</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_add_gene_names_xref_mars10x.html">mcell_add_gene_names_xref_mars10x()</a></code> </p>
        </td>
        <td><p>Create a gene name xref for heuristc map of mars/10x</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_add_gene_stat.html">mcell_add_gene_stat()</a></code> </p>
        </td>
        <td><p>mcell_gene_stat</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_add_mars_facs_data.html">mcell_add_mars_facs_data()</a></code> </p>
        </td>
        <td><p>Creates a new matrix object from a given one by adding FACS index sorting data to matrix metadata table</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_add_mc_from_graph.html">mcell_add_mc_from_graph()</a></code> </p>
        </td>
        <td><p>Compute metacell using a native implementation of a graph cover k-means-like approach</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_batch_stats.html">mcell_batch_stats()</a></code> </p>
        </td>
        <td><p>generate a batch stat table for a given single cell umi matrix</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_calc_one_batch_stats.html">mcell_calc_one_batch_stats()</a></code> </p>
        </td>
        <td><p>calc batch stats - essentially umi distribution</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_cgraph_norm_gcor.html">mcell_cgraph_norm_gcor()</a></code> </p>
        </td>
        <td><p>Computing gene-gene correlation normaized over a similarity graph</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_coclust_filt_by_k_deg.html">mcell_coclust_filt_by_k_deg()</a></code> </p>
        </td>
        <td><p>Return a filter (boolean vector) selecting only coclust edges that are nearly as frequent as a user defined K-nn parameter</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_coclust_from_graph_resamp.html">mcell_coclust_from_graph_resamp()</a></code> </p>
        </td>
        <td><p>Compute metacell using resampling iterations of graph cover k-means-like approach</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_compare_bulk.html">mcell_compare_bulk()</a></code> </p>
        </td>
        <td><p>Utility plot function to compare bulk expression of two batches/metadata factors</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_gen_atlas.html">mcell_gen_atlas()</a></code> </p>
        </td>
        <td><p>wrap up an atlas object from mc, mc2d and matrix ids</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_gen_cell_cor_gset.html">mcell_gen_cell_cor_gset()</a></code> </p>
        </td>
        <td><p>Compute a cell cell correlation matrix using features defined by a gene set</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_graph_pred_corr.html">mcell_graph_pred_corr()</a></code> </p>
        </td>
        <td><p>Compute predictive value of MC cover as correlation of MC averages of sc umis</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_gset_add_gene.html">mcell_gset_add_gene()</a></code> </p>
        </td>
        <td><p>adding genes to a gene set</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_gset_filter_cov.html">mcell_gset_filter_cov()</a></code> </p>
        </td>
        <td><p>gnereate/filter gene features from coverage threshold in gstat table</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_gset_filter_multi.html">mcell_gset_filter_multi()</a></code> </p>
        </td>
        <td><p>Select/filter gene features from using multiple statistics from the gstat table. All genes passing the selected thresholds are included</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_gset_filter_szcor.html">mcell_gset_filter_szcor()</a></code> </p>
        </td>
        <td><p>gnereate/filter gene features from statistics on correlation with umi count</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_gset_filter_varmean.html">mcell_gset_filter_varmean()</a></code> </p>
        </td>
        <td><p>gnereate/filter gene features from gstat normalized var/mean</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_gset_from_mc_markers.html">mcell_gset_from_mc_markers()</a></code> </p>
        </td>
        <td><p>Generate a gene set of markers from a metacell cover</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_gset_remove_clusts.html">mcell_gset_remove_clusts()</a></code> </p>
        </td>
        <td><p>add new gene set based on an existing one with filtering specific clusters</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_gset_split_by_dsmat.html">mcell_gset_split_by_dsmat()</a></code> </p>
        </td>
        <td><p>split gene set into several modules using clustering of genes by correlation over cells downsampled umi vector</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_import_multi_mars.html">mcell_import_multi_mars()</a></code> </p>
        </td>
        <td><p>Load a matrix from a MARS-seq multi-batch dataset. The scdb version of mcell_read_multi_scmat_mars</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_import_multi_scmat_10x.html">mcell_import_multi_scmat_10x()</a></code> </p>
        </td>
        <td><p>Load a matrix from a 10x multi-batch dataset. The scdb version of mcell_read_multi_scmat_10x</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_import_scmat_10x.html">mcell_import_scmat_10x()</a></code> </p>
        </td>
        <td><p>Load a matrix from a 10x dataset. The scdb version of mcell_read_multi_scmat_10x</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_import_scmat_tsv.html">mcell_import_scmat_tsv()</a></code> </p>
        </td>
        <td><p>Load a matrix from a simple dense table with genes in rows and cells in columns. First column is the gene name</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mat_ignore_cells.html">mcell_mat_ignore_cells()</a></code> </p>
        </td>
        <td><p>Generate a new matrix object with a given ignore cell list</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mat_ignore_genes.html">mcell_mat_ignore_genes()</a></code> </p>
        </td>
        <td><p>Generate a new matrix object with a given ignore gene list</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mat_ignore_small_cells.html">mcell_mat_ignore_small_cells()</a></code> </p>
        </td>
        <td><p>Generate a new matrix object after removing cells without enough umis</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mat_rpt_cor_anchors.html">mcell_mat_rpt_cor_anchors()</a></code> </p>
        </td>
        <td><p>Find and report naive gene gene correlation over umi matrix</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc2d_force_knn.html">mcell_mc2d_force_knn()</a></code> </p>
        </td>
        <td><p>Metacell layout using force directed projection of a low degree mc graph</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc2d_force_knn_on_cells.html">mcell_mc2d_force_knn_on_cells()</a></code> </p>
        </td>
        <td><p>Compute cells 2d coordinates based on the mc graph when the mc coordinates are supplied externally</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc2d_plot.html">mcell_mc2d_plot()</a></code> </p>
        </td>
        <td><p>Plot mc+cell graph using pre-defined mc colorization</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc2d_plot_by_factor.html">mcell_mc2d_plot_by_factor()</a></code> </p>
        </td>
        <td><p>Plot mc+cells using pre-defined mc colorization, breakdown by given metadata field (e.g. patient)</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc2d_plot_gene.html">mcell_mc2d_plot_gene()</a></code> </p>
        </td>
        <td><p>Plot the (log2) metacell footprint value of the selected gene on the 2d projection</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc2d_rotate.html">mcell_mc2d_rotate()</a></code> </p>
        </td>
        <td><p>Rotatae/invert projection</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc_add_annot.html">mcell_mc_add_annot()</a></code> </p>
        </td>
        <td><p>Update metacell annotation</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc_add_color.html">mcell_mc_add_color()</a></code> </p>
        </td>
        <td><p>Update metacells  colors</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc_cell_homogeneity.html">mcell_mc_cell_homogeneity()</a></code> </p>
        </td>
        <td><p>Compute cell homogeneity - the fraction of intra mc edges per cell</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc_coclust_confusion_mat.html">mcell_mc_coclust_confusion_mat()</a></code> </p>
        </td>
        <td><p>Compute confusion matrix on metacells using a coclustering object</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc_confusion_mat.html">mcell_mc_confusion_mat()</a></code> </p>
        </td>
        <td><p>Compute confusion matrix on metacells</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc_edge_density.html">mcell_mc_edge_density()</a></code> </p>
        </td>
        <td><p>for each MC we compute mean and variance of the intra-MC edge density</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc_export_tab.html">mcell_mc_export_tab()</a></code> </p>
        </td>
        <td><p>Output genes cell modules footprint matrix with metadata on the cell modules</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc_from_coclust_balanced.html">mcell_mc_from_coclust_balanced()</a></code> </p>
        </td>
        <td><p>build a metacell cover from co-clust data through filtering un-balanced edges
and running graph cover</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc_from_coclust_hc.html">mcell_mc_from_coclust_hc()</a></code> </p>
        </td>
        <td><p>build a metacell cover from a co-clust object using a simple hclust approach</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc_from_coclust_louv_sub.html">mcell_mc_from_coclust_louv_sub()</a></code> </p>
        </td>
        <td><p>build a metacell cover from a big co-clust using louvain clustering and metacell coverage within clusters</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc_hclust_confu.html">mcell_mc_hclust_confu()</a></code> </p>
        </td>
        <td><p>comput mc hierarchucal clustering using the normalized confusion matrix</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc_hierarchy.html">mcell_mc_hierarchy()</a></code> </p>
        </td>
        <td><p>identify super structure in an mc cover, based on hcluster of the confusion matrix</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc_match_graph.html">mcell_mc_match_graph()</a></code> </p>
        </td>
        <td><p>TEst if a graph object cover all cells in the mc</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc_plot_by_factor.html">mcell_mc_plot_by_factor()</a></code> </p>
        </td>
        <td><p>plot a heatmap of number of cells per metacell and metadata factor (e.g. patient, condition, sample etc.)</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc_plot_confusion.html">mcell_mc_plot_confusion()</a></code> </p>
        </td>
        <td><p>plot a metacel confusion matrix</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc_plot_gg.html">mcell_mc_plot_gg()</a></code> </p>
        </td>
        <td><p>Utility plot function to compare two genes based on mc_fp</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc_plot_hierarchy.html">mcell_mc_plot_hierarchy()</a></code> </p>
        </td>
        <td><p>plot super strucutre: super clust mc footprint, and selected genes</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc_plot_marks.html">mcell_mc_plot_marks()</a></code> </p>
        </td>
        <td><p>plot a marker heat map give a metacell object</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc_plot_submc_marks.html">mcell_mc_plot_submc_marks()</a></code> </p>
        </td>
        <td><p>plot a marker heat map for a subset of metacells - selecting relevant genes for separation</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc_plot_vgels.html">mcell_mc_plot_vgels()</a></code> </p>
        </td>
        <td><p>plot a "gel" like diagram showing expression of a gene of interest over metacells that are classified into types</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc_pred_corr.html">mcell_mc_pred_corr()</a></code> </p>
        </td>
        <td><p>Compute predictive value of MC cover as correlation of MC averages of sc umis</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc_reorder_hc.html">mcell_mc_reorder_hc()</a></code> </p>
        </td>
        <td><p>Reorder metacells using hierarchical clustering</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc_screen_outliers_1gene_fold.html">mcell_mc_screen_outliers_1gene_fold()</a></code> </p>
        </td>
        <td><p>Simple screen for outlier cells in a metacell cover, finding genes with overly high expression given their metacell mean</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc_split_by_color_group.html">mcell_mc_split_by_color_group()</a></code> </p>
        </td>
        <td><p>Splits input metacell object into sub-objects by color group, naming the new metacells &lt;mc_id&gt;_submc_&lt;group&gt;</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mc_split_filt.html">mcell_mc_split_filt()</a></code> </p>
        </td>
        <td><p>Split and filter metacells using dbscan and outlier gene detection</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_merge_mats.html">mcell_merge_mats()</a></code> </p>
        </td>
        <td><p>ncell_merge_mats: Merge two matrix using their ids in scdb. See scm_merge_mats for details on batch management and policies on missing genes</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mgraph_knn.html">mcell_mgraph_knn()</a></code> </p>
        </td>
        <td><p>Compute metacell manifod graph using the confusion matrix of balanced K-nn between individual cells projected on metacells</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_mgraph_logistic.html">mcell_mgraph_logistic()</a></code> </p>
        </td>
        <td><p>Compute metacell manifold graph using logistic distances and balanced K-nn</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_new_mc.html">mcell_new_mc()</a></code> </p>
        </td>
        <td><p>Generate a new metacell in scdb</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_new_mc2d.html">mcell_new_mc2d()</a></code> </p>
        </td>
        <td><p>Generate a new metacell in scdb</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_new_mc_mgraph.html">mcell_new_mc_mgraph()</a></code> </p>
        </td>
        <td><p>Generate a new metacell manifold graph object</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_new_mcatlas.html">mcell_new_mcatlas()</a></code> </p>
        </td>
        <td><p>Generate a new atlas in scdb</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_new_mctnetwork.html">mcell_new_mctnetwork()</a></code> </p>
        </td>
        <td><p>Generate a new network in scdb</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_pipe_mat2mc2d.html">mcell_pipe_mat2mc2d()</a></code> </p>
        </td>
        <td><p>Linear simple pipeline for turning a matrix to a MC and plotting std figs</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_plot_batch_stats.html">mcell_plot_batch_stats()</a></code> </p>
        </td>
        <td><p>plot batches stats</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_plot_cross_mc.html">mcell_plot_cross_mc()</a></code> </p>
        </td>
        <td><p>Plotting a matrix of co-occurences between two metacell covers of the same dataset</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_plot_gset_cor_mats.html">mcell_plot_gset_cor_mats()</a></code> </p>
        </td>
        <td><p>Plot gene set correlation matrices given a an scamt. See version using metacells for potentially more robust behavior. This is used to detemrine initial feature selectio (e.g. filtering biologically irrelevant gene modules)</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_plot_gstats.html">mcell_plot_gstats()</a></code> </p>
        </td>
        <td><p>plot gene/feature statistics</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_plot_outlier_heatmap.html">mcell_plot_outlier_heatmap()</a></code> </p>
        </td>
        <td><p>Plot and outlier heat map.</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_plot_umis_per_cell.html">mcell_plot_umis_per_cell()</a></code> </p>
        </td>
        <td><p>Plot histogram of total number of umis per cell in the umis matrix</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_proj_on_atlas.html">mcell_proj_on_atlas()</a></code> </p>
        </td>
        <td><p>Project a metacell object on a reference "atlas"</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_read_multi_scmat_10x.html">mcell_read_multi_scmat_10x()</a></code> </p>
        </td>
        <td><p>read multiple 10x umi matrices and merge them, based on a table defining the datasets. Field amp_batch_id from the table is added to the cell name to prevent cell names clashes.</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_read_multi_scmat_mars.html">mcell_read_multi_scmat_mars()</a></code> </p>
        </td>
        <td><p>read multiple MARS umi matrices and merge them, based on a table defining the datasets.</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_scmat_plot_cmp_kcor.html">mcell_scmat_plot_cmp_kcor()</a></code> </p>
        </td>
        <td><p>Analyze cell cell cor</p></td>
      </tr><tr><td>
          <p><code><a href="mcell_wgtmc_pred_corr.html">mcell_wgtmc_pred_corr()</a></code> </p>
        </td>
        <td><p>Compute predictive value of MC cover as correlation of weighted MC averages of sc umis</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="scdb-functions">scdb functions <a href="#scdb-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="scdb_add_cgraph.html">scdb_add_cgraph()</a></code> </p>
        </td>
        <td><p>scdb_add_cgraph - add cgraph to the DB and cahce</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_add_coclust.html">scdb_add_coclust()</a></code> </p>
        </td>
        <td><p>scdb_add_coclust - add coclust to the DB and cahce</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_add_gene_names_xref.html">scdb_add_gene_names_xref()</a></code> </p>
        </td>
        <td><p>scdb_add_gene_names_xref - add a gene name xref tab to the DB - will save it and cache</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_add_gset.html">scdb_add_gset()</a></code> </p>
        </td>
        <td><p>scdb_add_gset - add gset to the DB and cahce</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_add_gstat.html">scdb_add_gstat()</a></code> </p>
        </td>
        <td><p>scdb_add_gstat - add gstat to the DB and cahce</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_add_mat.html">scdb_add_mat()</a></code> </p>
        </td>
        <td><p>scdb_add_mat - add amatrix to the DB - will save it and cache</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_add_mc.html">scdb_add_mc()</a></code> </p>
        </td>
        <td><p>scdb_add_mc - add mc to the DB and cahce</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_add_mc2d.html">scdb_add_mc2d()</a></code> </p>
        </td>
        <td><p>scdb_add_mc2d - add mc2d to the DB and cahce</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_add_mcatlas.html">scdb_add_mcatlas()</a></code> </p>
        </td>
        <td><p>scdb_add_mcatlas - add mcatlas to the DB and cahce</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_add_mctnetwork.html">scdb_add_mctnetwork()</a></code> </p>
        </td>
        <td><p>scdb_add_mctnetwork - add mctnetwork to the DB and cahce</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_add_mgraph.html">scdb_add_mgraph()</a></code> </p>
        </td>
        <td><p>scdb_add_mgraph - add mgraph to the DB and cahce</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_cgraph.html">scdb_cgraph()</a></code> </p>
        </td>
        <td><p>scdb_cgraph - get a cgraph object</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_coclust.html">scdb_coclust()</a></code> </p>
        </td>
        <td><p>scdb_coclust - get a coclust object</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_del_cgraph.html">scdb_del_cgraph()</a></code> </p>
        </td>
        <td><p>scdb_del_cgraph - del cgraph from the DB and cahce</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_del_coclust.html">scdb_del_coclust()</a></code> </p>
        </td>
        <td><p>scdb_del_coclust - del coclust from the DB and cahce</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_del_gene_names_xref.html">scdb_del_gene_names_xref()</a></code> </p>
        </td>
        <td><p>scdb_del_gene_names_xref - remove a gene names xref from the DB (not just the cache!)</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_del_gset.html">scdb_del_gset()</a></code> </p>
        </td>
        <td><p>scdb_del_gset - del gset from the DB and cahce</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_del_gstat.html">scdb_del_gstat()</a></code> </p>
        </td>
        <td><p>scdb_del_gstat - del gstat from the DB and cahce</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_del_mat.html">scdb_del_mat()</a></code> </p>
        </td>
        <td><p>scdb_del_mat - remove a matrix from the DB (not just the cache!)</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_del_mc.html">scdb_del_mc()</a></code> </p>
        </td>
        <td><p>scdb_del_mc - del mc from the DB and cahce</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_del_mc2d.html">scdb_del_mc2d()</a></code> </p>
        </td>
        <td><p>scdb_del_mc2d - del mc2d from the DB and cahce</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_del_mcatlas.html">scdb_del_mcatlas()</a></code> </p>
        </td>
        <td><p>scdb_del_mcatlas - del mcatlas from the DB and cahce</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_del_mctnetwork.html">scdb_del_mctnetwork()</a></code> </p>
        </td>
        <td><p>scdb_del_mctnetwork - del mctnetwork from the DB and cahce</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_del_mgraph.html">scdb_del_mgraph()</a></code> </p>
        </td>
        <td><p>scdb_del_mgraph - del mgraph from the DB and cahce</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_gene_names_xref.html">scdb_gene_names_xref()</a></code> </p>
        </td>
        <td><p>scdb_gene_names_xref - get gene names convertor table from db</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_gset.html">scdb_gset()</a></code> </p>
        </td>
        <td><p>scdb_gset - get a gene set</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_gstat.html">scdb_gstat()</a></code> </p>
        </td>
        <td><p>scdb_gstat - get a gstat data frame. If it is missing and the id match an existing matrix, a gstat will be gerated for this matrix and added to scdb</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_init.html">scdb_init()</a></code> </p>
        </td>
        <td><p>Initializing scdb</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_is_valid.html">scdb_is_valid()</a></code> </p>
        </td>
        <td><p>Testing is scdb is initialized</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_ls.html">scdb_ls()</a></code> </p>
        </td>
        <td><p>List all object of a given type from the current scdb</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_ls_loaded.html">scdb_ls_loaded()</a></code> </p>
        </td>
        <td><p>scdb_ls_loaded - list loaded object of a certain type</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_mat.html">scdb_mat()</a></code> </p>
        </td>
        <td><p>scdb_mat - get matrix from db, load it if needed</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_mc.html">scdb_mc()</a></code> </p>
        </td>
        <td><p>scdb_mc - get a mc object</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_mc2d.html">scdb_mc2d()</a></code> </p>
        </td>
        <td><p>scdb_mc2d - get a mc2d object</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_mcatlas.html">scdb_mcatlas()</a></code> </p>
        </td>
        <td><p>scdb_mcatlas - get a mcatlas object</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_mctnetwork.html">scdb_mctnetwork()</a></code> </p>
        </td>
        <td><p>scdb_mctnetwork - get a mctnetwork object</p></td>
      </tr><tr><td>
          <p><code><a href="scdb_mgraph.html">scdb_mgraph()</a></code> </p>
        </td>
        <td><p>scdb_mgraph - get a mgraph object</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="scm-functions">scm functions <a href="#scm-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="scm_add_mars_facs_data.html">scm_add_mars_facs_data()</a></code> </p>
        </td>
        <td><p>Adds FACS index sorting data to matrix metadata table</p></td>
      </tr><tr><td>
          <p><code><a href="scm_downsamp.html">scm_downsamp()</a></code> </p>
        </td>
        <td><p>downsampl</p></td>
      </tr><tr><td>
          <p><code><a href="scm_export_mat_to_sce.html">scm_export_mat_to_sce()</a></code> </p>
        </td>
        <td><p>Export a mat object (umi matrix and metadata table) to a SingleCellExperiment object</p></td>
      </tr><tr><td>
          <p><code><a href="scm_gene_stat.html">scm_gene_stat()</a></code> </p>
        </td>
        <td><p>Calculate basic statistics on a matrix</p></td>
      </tr><tr><td>
          <p><code><a href="scm_ignore_cells.html">scm_ignore_cells()</a></code> </p>
        </td>
        <td><p>Set ignored (i.e. blacklisted) cells</p></td>
      </tr><tr><td>
          <p><code><a href="scm_ignore_genes.html">scm_ignore_genes()</a></code> </p>
        </td>
        <td><p>Set ignored (i.e. blacklisted) genes</p></td>
      </tr><tr><td>
          <p><code><a href="scm_import_sce_to_mat.html">scm_import_sce_to_mat()</a></code> </p>
        </td>
        <td><p>Import a umi count matrix with metadata per cell from a SingleCellExperiment objectto a scmat object  to a SingleCellExperiment object</p></td>
      </tr><tr><td>
          <p><code><a href="scm_merge_mats.html">scm_merge_mats()</a></code> </p>
        </td>
        <td><p>Merge multiple single cell matrix object.</p></td>
      </tr><tr><td>
          <p><code><a href="scm_new_matrix.html">scm_new_matrix()</a></code> </p>
        </td>
        <td><p>Constract a tgScMat</p></td>
      </tr><tr><td>
          <p><code><a href="scm_sub_mat.html">scm_sub_mat()</a></code> </p>
        </td>
        <td><p>Extract sub-matrix. This return a matrix object on a subset of the genes and cells.</p></td>
      </tr><tr><td>
          <p><code><a href="scm_which_downsamp_n.html">scm_which_downsamp_n()</a></code> </p>
        </td>
        <td><p>Determine recommended downsampling depth.</p></td>
      </tr><tr><td>
          <p><code><a href="scmat_read_scmat_10x.html">scmat_read_scmat_10x()</a></code> </p>
        </td>
        <td><p>Read a matrix from the output of a 10x run. Batches can be stripped from the cell identifier if in BARCODE-LANE format.</p></td>
      </tr><tr><td>
          <p><code><a href="scmat_read_scmat_10x_custom.html">scmat_read_scmat_10x_custom()</a></code> </p>
        </td>
        <td><p>Read a custom count matrix from the output of a 10x run. Batches can be stripped from the cell identifier if in BARCODE-LANE format.</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="metacell-functions">metacell functions <a href="#metacell-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="mc_colorize.html">mc_colorize()</a></code> </p>
        </td>
        <td><p>colorize metacell using a set of prefered markers and their colors</p></td>
      </tr><tr><td>
          <p><code><a href="mc_colorize_default.html">mc_colorize_default()</a></code> </p>
        </td>
        <td><p>colorize metacell using an ugly default color spectrum, or a user supplied one</p></td>
      </tr><tr><td>
          <p><code><a href="mc_colorize_from_ref_mc.html">mc_colorize_from_ref_mc()</a></code> </p>
        </td>
        <td><p>colorize metacell by projecting colors from another metacell on a similar (not identical) set of cells</p></td>
      </tr><tr><td>
          <p><code><a href="mc_colorize_sup_hierarchy.html">mc_colorize_sup_hierarchy()</a></code> </p>
        </td>
        <td><p>colorize metacells using a set of super MCs derived by hclust, colored according to a user defined table</p></td>
      </tr><tr><td>
          <p><code><a href="mc_compute_cov_gc.html">mc_compute_cov_gc()</a></code> </p>
        </td>
        <td><p>Compute fraction of non zero expressing cells per gene and mc</p></td>
      </tr><tr><td>
          <p><code><a href="mc_compute_e_gc.html">mc_compute_e_gc()</a></code> </p>
        </td>
        <td><p>Compute metacell absolute mean umi per cell</p></td>
      </tr><tr><td>
          <p><code><a href="mc_compute_fp.html">mc_compute_fp()</a></code> </p>
        </td>
        <td><p>Compute metacell gene footprint</p></td>
      </tr><tr><td>
          <p><code><a href="mc_compute_n_bc.html">mc_compute_n_bc()</a></code> </p>
        </td>
        <td><p>Compute distribution of cells over batches and metacell</p></td>
      </tr><tr><td>
          <p><code><a href="mc_compute_outlier_fc.html">mc_compute_outlier_fc()</a></code> </p>
        </td>
        <td><p>Compute log fold change expression of each gene given its regularized metacell expression</p></td>
      </tr><tr><td>
          <p><code><a href="mc_reorder.html">mc_reorder()</a></code> </p>
        </td>
        <td><p>Reorder metacell data given defined order</p></td>
      </tr><tr><td>
          <p><code><a href="mc_restrict.html">mc_restrict()</a></code> </p>
        </td>
        <td><p>Create a metacell object on a subset of the MCs, with all other cells becoming outliers. There is no re-normalization of mc_fp.</p></td>
      </tr><tr><td>
          <p><code><a href="mc_set_outlier_mc.html">mc_set_outlier_mc()</a></code> </p>
        </td>
        <td><p>Move all cels from specific metacells to the outliers</p></td>
      </tr><tr><td>
          <p><code><a href="mc_update_stats.html">mc_update_stats()</a></code> </p>
        </td>
        <td><p>Compute stats over metacell and update the object</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="gene-set-functions">gene set functions <a href="#gene-set-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="gset_add_genes.html">gset_add_genes()</a></code> </p>
        </td>
        <td><p>Add specific genes to the set</p></td>
      </tr><tr><td>
          <p><code><a href="gset_get_feat_mat.html">gset_get_feat_mat()</a></code> </p>
        </td>
        <td><p>extract umi matrix for the genes in the set, possibly donwsampling</p></td>
      </tr><tr><td>
          <p><code><a href="gset_get_genes.html">gset_get_genes()</a></code> </p>
        </td>
        <td><p>Get genes of one set</p></td>
      </tr><tr><td>
          <p><code><a href="gset_import_table.html">gset_import_table()</a></code> </p>
        </td>
        <td><p>Import gene set for a text table</p></td>
      </tr><tr><td>
          <p><code><a href="gset_new_gset.html">gset_new_gset()</a></code> </p>
        </td>
        <td><p>Generating a new gset</p></td>
      </tr><tr><td>
          <p><code><a href="gset_new_restrict_gset.html">gset_new_restrict_gset()</a></code> </p>
        </td>
        <td><p>Generate a new gene set from an existing one, filtered by a list of genes</p></td>
      </tr><tr><td>
          <p><code><a href="gset_new_restrict_nms.html">gset_new_restrict_nms()</a></code> </p>
        </td>
        <td><p>Generate a new gene set from an existing one, filtered by a list of genes</p></td>
      </tr><tr><td>
          <p><code><a href="gset_write_table.html">gset_write_table()</a></code> </p>
        </td>
        <td><p>Exprt gene set to a table</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="classes">classes <a href="#classes" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="tgCellGraph-class.html">tgCellGraph-class</a></code> </p>
        </td>
        <td><p>Cell graph</p></td>
      </tr><tr><td>
          <p><code><a href="tgCoClust-class.html">tgCoClust-class</a></code> </p>
        </td>
        <td><p>Metacell colustering object</p></td>
      </tr><tr><td>
          <p><code><a href="tgGeneSets-class.html">tgGeneSets-class</a></code> </p>
        </td>
        <td><p>Gene sets interface</p></td>
      </tr><tr><td>
          <p><code><a href="tgMC2D-class.html">tgMC2D-class</a></code> </p>
        </td>
        <td><p>Meta cell 2d projection</p></td>
      </tr><tr><td>
          <p><code><a href="tgMCAtlas-class.html">tgMCAtlas-class</a></code> </p>
        </td>
        <td><p>wrap up an atlas object from mc, mc2d and matrix ids</p></td>
      </tr><tr><td>
          <p><code><a href="tgMCCov-class.html">tgMCCov-class</a></code> </p>
        </td>
        <td><p>Meta cell cover</p></td>
      </tr><tr><td>
          <p><code><a href="tgMCManifGraph-class.html">tgMCManifGraph-class</a></code> </p>
        </td>
        <td><p>manifold graph structure over a metacell object</p></td>
      </tr><tr><td>
          <p><code><a href="tgMCTNetwork-class.html">tgMCTNetwork-class</a></code> </p>
        </td>
        <td><p>temporal netwrok over metacells</p></td>
      </tr><tr><td>
          <p><code><a href="tgScMat-class.html">tgScMat-class</a></code> </p>
        </td>
        <td><p>Single cell RNA-seq matrix</p></td>
      </tr><tr><td>
          <p><code><a href="initialize-tgCellGraph-method.html">initialize(<i>&lt;tgCellGraph&gt;</i>)</a></code> </p>
        </td>
        <td><p>This constructs graph object from a data frame on columns col1,col2, weight, (the format returened by tgs_cor_graph). Note this is not intended to serve natively complex graph algs etc, but is just a container for use by scdb and operaitons on scmats (creating a graph) and mcell (using the graph to compute graph covers)</p></td>
      </tr><tr><td>
          <p><code><a href="initialize-tgCoClust-method.html">initialize(<i>&lt;tgCoClust&gt;</i>)</a></code> </p>
        </td>
        <td><p>Construct a coclust object</p></td>
      </tr><tr><td>
          <p><code><a href="initialize-tgGeneSets-method.html">initialize(<i>&lt;tgGeneSets&gt;</i>)</a></code> </p>
        </td>
        <td><p>tgGeneSets public constructor</p></td>
      </tr><tr><td>
          <p><code><a href="initialize-tgMC2D-method.html">initialize(<i>&lt;tgMC2D&gt;</i>)</a></code> </p>
        </td>
        <td><p>Construct a meta cell 2D embedding</p></td>
      </tr><tr><td>
          <p><code><a href="initialize-tgMCAtlas-method.html">initialize(<i>&lt;tgMCAtlas&gt;</i>)</a></code> </p>
        </td>
        <td><p>Construct a meta cell reference atlas</p></td>
      </tr><tr><td>
          <p><code><a href="initialize-tgMCCov-method.html">initialize(<i>&lt;tgMCCov&gt;</i>)</a></code> </p>
        </td>
        <td><p>Construct a meta cell object</p></td>
      </tr><tr><td>
          <p><code><a href="initialize-tgMCManifGraph-method.html">initialize(<i>&lt;tgMCManifGraph&gt;</i>)</a></code> </p>
        </td>
        <td><p>Construct a meta cell manifold graph</p></td>
      </tr><tr><td>
          <p><code><a href="initialize-tgMCTNetwork-method.html">initialize(<i>&lt;tgMCTNetwork&gt;</i>)</a></code> </p>
        </td>
        <td><p>Construct a meta cell time network</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="utils">utils <a href="#utils" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="plot_color_bar.html">plot_color_bar()</a></code> </p>
        </td>
        <td><p>Plot a color bar with values</p></td>
      </tr><tr><td>
          <p><code><a href="port_cor.html">port_cor()</a></code> </p>
        </td>
        <td><p>port_cor
matrix correlation wrap. Parameter selects which function to use, cor or tgs_cor. Tgs_cor can be more efficien if available</p></td>
      </tr><tr><td>
          <p><code><a href="allrow_cor.html">allrow_cor()</a></code> </p>
        </td>
        <td><p>computing correlations between all rows in two matrices</p></td>
      </tr><tr><td>
          <p><code><a href="rescale_sparse_mat_cols.html">rescale_sparse_mat_cols()</a></code> </p>
        </td>
        <td><p>Efficient version for scaling all columns in a sparse matrix</p></td>
      </tr><tr><td>
          <p><code><a href="tgs_cor_graph.html">tgs_cor_graph()</a></code> </p>
        </td>
        <td><p>wrapping tgs functions to compute balanced graph from a matrix</p></td>
      </tr><tr><td>
          <p><code><a href="scfigs_dir.html">scfigs_dir()</a></code> </p>
        </td>
        <td><p>Generate a standard figure dir name igven and object and figure type</p></td>
      </tr><tr><td>
          <p><code><a href="scfigs_fn.html">scfigs_fn()</a></code> </p>
        </td>
        <td><p>Generate a standard figure name igven and object and figure type</p></td>
      </tr><tr><td>
          <p><code><a href="scfigs_init.html">scfigs_init()</a></code> </p>
        </td>
        <td><p>scgfig_set_base - set base directory for metacell figures, creates it if it doesn't exist.</p></td>
      </tr></tbody></table></div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="http://compgenomics.weizmann.ac.il/tanay/" class="external-link">Amos Tanay</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

      </footer></div>

  


  

  </body></html>

