<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Metacell layout using force directed projection of a low degree mc graph — mcell_mc2d_force_knn • metacell</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Metacell layout using force directed projection of a low degree mc graph — mcell_mc2d_force_knn"><meta property="og:description" content="Metacell layout using force directed projection of a low degree mc graph"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">metacell</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/a-basic_pbmc8k.html">Usage</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li class="dropdown-header">Core</li>
    <li>
      <a href="../articles/a-basic_pbmc8k.html">Running metacell analysis: guided tutorial on 8K PBMCs</a>
    </li>
    <li>
      <a href="../articles/b-geneset_by_anchor_pbmc8k.html">Supervised filtering of feature genes: guided tutorial on 8K PBMCs</a>
    </li>
    <li>
      <a href="../articles/c-metacell_annotation.html">Annotation of metacells</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Misc</li>
    <li>
      <a href="../articles/d-amphimedon.html">Analyzing whole-organism scRNA-seq data with metacell</a>
    </li>
    <li>
      <a href="../articles/e-data_convert.html">UMI matrix conversion to standard formats</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a></a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Metacell layout using force directed projection of a low degree mc graph</h1>
    <small class="dont-index">Source: <a href="https://github.com/tanaylab/metacell/blob/HEAD/R/mc2d_force_knn.r" class="external-link"><code>R/mc2d_force_knn.r</code></a></small>
    <div class="hidden name"><code>mcell_mc2d_force_knn.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Metacell layout using force directed projection of a low degree mc graph</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">mcell_mc2d_force_knn</span><span class="op">(</span>
  <span class="va">mc2d_id</span>,
  <span class="va">mc_id</span>,
  <span class="va">graph_id</span>,
  mgraph_id <span class="op">=</span> <span class="cn">NULL</span>,
  ignore_mismatch <span class="op">=</span> <span class="cn">F</span>,
  symmetrize <span class="op">=</span> <span class="cn">F</span>,
  ignore_edges <span class="op">=</span> <span class="cn">NULL</span>,
  feats_gset <span class="op">=</span> <span class="cn">NULL</span>,
  feats_exclude <span class="op">=</span> <span class="cn">NULL</span>,
  graph_parametric <span class="op">=</span> <span class="cn">F</span>,
  logist_loc <span class="op">=</span> <span class="fl">1</span>,
  logist_scale <span class="op">=</span> <span class="fl">0.2</span>,
  logist_eps <span class="op">=</span> <span class="fl">0.00004</span>,
  max_d_fold <span class="op">=</span> <span class="fl">3</span>,
  use_umap_for_2d_pos <span class="op">=</span> <span class="cn">F</span>,
  umap_mgraph <span class="op">=</span> <span class="cn">F</span>,
  uconf <span class="op">=</span> <span class="cn">NULL</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>mc2d_id</dt>
<dd><p>2d object to add</p></dd>
<dt>mc_id</dt>
<dd><p>meta cell id to work with</p></dd>
<dt>graph_id</dt>
<dd><p>graph_id of the similarity graph on cells from the metacell. This should always be specified, even if an mgraph is supplied since we use the obejct to position cells releative to metacels</p></dd>
<dt>mgraph_id</dt>
<dd><p>mgraph_id of the metacell graph to use. If this is null the function will compute an mgraph itself using the cell graph</p></dd>
<dt>ignore_mismatch</dt>
<dd><p>determine if mismatch in mc names are to be ignored.</p></dd>
<dt>feats_gset</dt>
<dd><p>gene set name for use in parametric graph and/or umap 2d projection</p></dd>
<dt>feats_exclude</dt>
<dd><p>list of genes to exclude from the features gene set</p></dd>
<dt>graph_parametric</dt>
<dd><p>if this is True, mc edges will be determined using parametric distances and not the cells k-nn graph.</p></dd>
<dt>logist_loc</dt>
<dd><p>the location parametr of the logistic function used to determine parametric distances between metacelles</p></dd>
<dt>logist_scale</dt>
<dd><p>the lscale parametr of the logistic function used to determine parametric distances between metacelles</p></dd>
<dt>symetrize</dt>
<dd><p>should the mc confusion matrix be symmetrized before computing layout?</p></dd>
</dl></div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="http://compgenomics.weizmann.ac.il/tanay/" class="external-link">Amos Tanay</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

      </footer></div>

  


  

  </body></html>

