<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Supervised filtering of feature genes: guided tutorial on 8K PBMCs â€¢ metacell</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Supervised filtering of feature genes: guided tutorial on 8K PBMCs">
<meta property="og:description" content="metacell">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">metacell</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/a-basic_pbmc8k.html">Usage</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">Core</li>
    <li>
      <a href="../articles/a-basic_pbmc8k.html">Running metacell analysis: guided tutorial on 8K PBMCs</a>
    </li>
    <li>
      <a href="../articles/b-geneset_by_anchor_pbmc8k.html">Supervised filtering of feature genes: guided tutorial on 8K PBMCs</a>
    </li>
    <li>
      <a href="../articles/c-metacell_annotation.html">Annotation of metacells</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Misc</li>
    <li>
      <a href="../articles/d-amphimedon.html">Analyzing whole-organism scRNA-seq data with metacell</a>
    </li>
    <li>
      <a href="../articles/e-data_convert.html">UMI matrix conversion to standard formats</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a></a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Supervised filtering of feature genes: guided tutorial on 8K PBMCs</h1>
                        <h4 data-toc-skip class="author">Yaniv Lubling</h4>
            
            <h4 data-toc-skip class="date">2018-10-10</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/tanaylab/metacell/blob/HEAD/vignettes/b-geneset_by_anchor_pbmc8k.Rmd" class="external-link"><code>vignettes/b-geneset_by_anchor_pbmc8k.Rmd</code></a></small>
      <div class="hidden name"><code>b-geneset_by_anchor_pbmc8k.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This note walks through the common scenario of disabling specific gene modules from affecting the partitioning of cells into metacells. In many applications we would want to mask technical effects of stress (e.g. heat shock) from affecting the partitioning, or to mask cell cycle related genes in order to get metacells that represent cell types/states and not stages at the cell cycle.</p>
<p>We'll start with finding genes that are correlated to a few cell cycle and stress halmark genes, then generate gene correlation heatmaps that will help to further filter the genes, and lastly show how to create metacells that are not affected by these genes.</p>
<p>We use here the small 8K PBMC dataset that is used in the <strong>Running metacell analysis: guided tutorial on 8K PBMCs</strong> vignette, please make sure to run that vignette before executing this one, since we'll use several objects produced there.</p>
</div>
<div class="section level2">
<h2 id="setting-up">Setting up<a class="anchor" aria-label="anchor" href="#setting-up"></a>
</h2>
<p>We start by loading the library and initializating the database and the figures directories:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/tanaylab/metacell" class="external-link">"metacell"</a></span><span class="op">)</span>

<span class="fu"><a href="../reference/scdb_init.html">scdb_init</a></span><span class="op">(</span><span class="st">"testdb/"</span>, force_reinit<span class="op">=</span><span class="cn">T</span><span class="op">)</span>
<span class="co">#&gt; initializing scdb to testdb/</span>

<span class="va">figs_dir</span> <span class="op">=</span> <span class="st">"figs"</span>
<span class="fu"><a href="../reference/scfigs_init.html">scfigs_init</a></span><span class="op">(</span><span class="va">figs_dir</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="creating-the-gene-correlation-matrix">Creating the gene correlation matrix<a class="anchor" aria-label="anchor" href="#creating-the-gene-correlation-matrix"></a>
</h2>
<p>We're going to generate a gene-gene correlation matrix using the umi matrix generated in the <strong>Running metacell analysis: guided tutorial on 8K PBMCs</strong> vignette. The genes will then be filtered by thresholding their correlation with the user supplied genes (termed anchor genes). In this vignette we'll build a set of cell cycle and stress related genes. So let's define these gene anchors, and some additional variables:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">genes_anchors</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'PCNA'</span>, <span class="st">'TOP2A'</span>, <span class="st">'TXN'</span>, <span class="st">'HSP90AB1'</span>, <span class="st">'FOS'</span><span class="op">)</span>
<span class="va">tab_fn</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">figs_dir</span>, <span class="st">"lateral_gmods.txt"</span>, sep<span class="op">=</span><span class="st">"/"</span><span class="op">)</span>
<span class="va">gset_nm</span> <span class="op">=</span> <span class="st">"lateral"</span></code></pre></div>
<p><em>tab_fn</em> is the output file name of the correlation matrix, and <em>gset_nm</em> is the id of the gene set that will be generated.</p>
<p>Now we'll call the function that computes the gene-gene correlation matrix:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/mcell_mat_rpt_cor_anchors.html">mcell_mat_rpt_cor_anchors</a></span><span class="op">(</span>mat_id<span class="op">=</span><span class="st">"test"</span>, gene_anchors <span class="op">=</span> <span class="va">genes_anchors</span>, cor_thresh <span class="op">=</span> <span class="fl">0.1</span>, gene_anti <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span>, tab_fn <span class="op">=</span> <span class="va">tab_fn</span>, sz_cor_thresh <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span></code></pre></div>
<p>The function computes the correlations of all genes in the umi matrix with the gene anchors after downsampling the umi counts. It then filters genes that their correlation to the anchor genes is above <em>cor_thresh</em>. Genes can also be filtered based on their correlation to a set of negative control genes, that can be specified in the <em>gene_anti</em> parameter. In this case, genes that their top correlated gene is a negative control and not one of the anchor genes are discarded. If the <em>sz_cor_thresh</em> is not <em>NA</em>, genes with correlation to the cells total umi counts above <em>sz_cor_thresh</em> are also selected. In our case, we expect some cell cycle genes to be correlated with cell size, so we specify a positive threshold.</p>
<p>Let's read the filtered gene correlation matrix and examine it:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gcor_mat</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="va">tab_fn</span>, header<span class="op">=</span><span class="cn">T</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gcor_mat</span><span class="op">)</span>
<span class="co">#&gt;                 sz_cor         PCNA        TOP2A        TXN    HSP90AB1</span>
<span class="co">#&gt; AC023590.1 0.046756633  0.010617491 -0.003780935 0.18146371 -0.00416468</span>
<span class="co">#&gt; ACTB       0.250543976  0.059445816  0.111477842 0.13386438 -0.05907979</span>
<span class="co">#&gt; ACTG1      0.213020480  0.045909657  0.094956945 0.04625107  0.04491417</span>
<span class="co">#&gt; AGTRAP     0.001536675  0.010654896 -0.009926729 0.07963326 -0.07378579</span>
<span class="co">#&gt; AHNAK      0.057235736 -0.005677824  0.002048814 0.09009918 -0.04284507</span>
<span class="co">#&gt; AHR        0.054036998 -0.005936477 -0.007005925 0.05127561 -0.04556298</span>
<span class="co">#&gt;                    FOS        max neg_max</span>
<span class="co">#&gt; AC023590.1 -0.01856910 0.18146371       0</span>
<span class="co">#&gt; ACTB        0.13500093 0.13500093       0</span>
<span class="co">#&gt; ACTG1      -0.08386522 0.09495695       0</span>
<span class="co">#&gt; AGTRAP      0.14986172 0.14986172       0</span>
<span class="co">#&gt; AHNAK       0.13974787 0.13974787       0</span>
<span class="co">#&gt; AHR         0.10690160 0.10690160       0</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">gcor_mat</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 560   8</span></code></pre></div>
<p>The row names are genes, and the columns are the correlation to the total umi counts (<em>sz_cor</em>), the correlations to the anchor genes and their maximal one (<em>max</em>). Since we didn't specify negative anchor genes in this example the maximal correlation to the negative anchors is zero (<em>max_neg</em>).</p>
<p>Now we'll add a gene set to the database, comprised of the genes in the filtered matrix:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">foc_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">gcor_mat</span><span class="op">[</span>, <span class="va">genes_anchors</span><span class="op">]</span>, <span class="fl">1</span>, <span class="va">which.max</span><span class="op">)</span>

<span class="va">gset</span> <span class="op">=</span> <span class="fu"><a href="../reference/gset_new_gset.html">gset_new_gset</a></span><span class="op">(</span>sets <span class="op">=</span> <span class="va">foc_genes</span>, desc <span class="op">=</span> <span class="st">"Cell cycle and stress correlated genes"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">gset</span><span class="op">)</span>
<span class="co">#&gt; An object of class "tgGeneSets"</span>
<span class="co">#&gt; Slot "description":</span>
<span class="co">#&gt; [1] "Cell cycle and stress correlated genes"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Slot "set_names":</span>
<span class="co">#&gt; [1] 3 5 2 4 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Slot "gene_set":</span>
<span class="co">#&gt;    AC023590.1          ACTB         ACTG1        AGTRAP         AHNAK </span>
<span class="co">#&gt;             3             5             2             5             5 </span>
<span class="co">#&gt;           AHR          AIF1       ALDH1A1         ALDH2       ALDH3B1 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;         ALDOA       ALOX5AP       ANKRD28         ANPEP         ANXA1 </span>
<span class="co">#&gt;             5             3             5             5             5 </span>
<span class="co">#&gt;         ANXA2         ANXA5          AOAH         AP1S2         AP2S1 </span>
<span class="co">#&gt;             3             5             5             5             5 </span>
<span class="co">#&gt;         APLP2           APP      ARHGEF40         ARL5B     ARL6IP1P2 </span>
<span class="co">#&gt;             5             3             5             5             2 </span>
<span class="co">#&gt;        ARRDC4         ASAH1         ASF1B         ASGR1          ASIP </span>
<span class="co">#&gt;             5             5             2             5             3 </span>
<span class="co">#&gt;          ATF3          ATG3         ATOX1    ATP2B1-AS1         ATP5E </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;        ATP5G2         ATP5I       ATP6V0B      ATP6V0D1      ATP6V0E1 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;       ATP6V1A      ATP6V1B2       ATP6V1F         BACH1         BCAT1 </span>
<span class="co">#&gt;             5             5             5             5             3 </span>
<span class="co">#&gt;        BCL2A1         BIRC5       BLOC1S1         BLVRB        BNIP3L </span>
<span class="co">#&gt;             5             2             5             5             5 </span>
<span class="co">#&gt;          BRI3          BST1      C10orf11      C12orf75       C14orf2 </span>
<span class="co">#&gt;             5             5             5             3             5 </span>
<span class="co">#&gt;      C19orf38      C19orf47      C1orf162      C20orf27       C4orf48 </span>
<span class="co">#&gt;             5             2             5             5             5 </span>
<span class="co">#&gt;         C5AR1       C7orf73       C9orf72         CALM2         CAMK1 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;          CAPG        CAPZA2        CARD16         CARS2         CASP1 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;      CATSPER1       CCDC189        CCDC50       CCDC71L       CCDC88A </span>
<span class="co">#&gt;             5             3             3             2             5 </span>
<span class="co">#&gt;          CCL3        CCL3L3         CCNA2         CCNL1          CCNY </span>
<span class="co">#&gt;             5             5             2             5             5 </span>
<span class="co">#&gt;          CCR1          CCR9          CD14         CD163        CD300E </span>
<span class="co">#&gt;             5             2             5             5             5 </span>
<span class="co">#&gt;       CD300LF         CD302          CD33          CD36           CD4 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;          CD44          CD63          CD70          CD83          CD86 </span>
<span class="co">#&gt;             5             5             2             5             5 </span>
<span class="co">#&gt;          CD93           CDA         CDC42         CDCA8          CDK1 </span>
<span class="co">#&gt;             5             5             5             2             2 </span>
<span class="co">#&gt;          CDK2        CDKN2C         CDKN3          CDR2         CEBPB </span>
<span class="co">#&gt;             2             2             2             2             5 </span>
<span class="co">#&gt;         CEBPD         CECR1         CENPE         CENPF         CENPN </span>
<span class="co">#&gt;             5             5             2             2             2 </span>
<span class="co">#&gt;         CENPP         CENPU           CFD           CFP        CHMP1B </span>
<span class="co">#&gt;             2             2             5             5             5 </span>
<span class="co">#&gt;          CHP1          CIB2         CKAP4          CKLF       CLEC10A </span>
<span class="co">#&gt;             5             3             5             5             3 </span>
<span class="co">#&gt;       CLEC12A        CLEC4A        CLEC4C        CLEC4E        CLEC7A </span>
<span class="co">#&gt;             5             5             3             5             5 </span>
<span class="co">#&gt;         CMTM6         CNPY3          COMT        CORO1A         COTL1 </span>
<span class="co">#&gt;             5             5             3             2             5 </span>
<span class="co">#&gt;        COX6B1         COX7B        CPPED1          CPVL         CREB5 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;         CREG1         CRTAP          CRYM         CSF3R          CST3 </span>
<span class="co">#&gt;             5             5             3             5             5 </span>
<span class="co">#&gt;          CSTA    CTB-61M7.2  CTC-510F12.4  CTD-3252C9.4          CTSB </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;          CTSC          CTSD          CTSH          CTSS          CTSZ </span>
<span class="co">#&gt;             3             5             5             5             5 </span>
<span class="co">#&gt;         CXCL2         CXCL8          CYBA          CYBB        CYP1B1 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;        DAZAP2         DERL3         DMXL2      DNASE1L3          DOK3 </span>
<span class="co">#&gt;             5             3             5             3             5 </span>
<span class="co">#&gt;          DPM2          DPYD        DPYSL2        DSTNP2         DUSP1 </span>
<span class="co">#&gt;             2             5             5             2             5 </span>
<span class="co">#&gt;         DUSP6        DYNLL1       EFCAB11          EGR1       EHBP1L1 </span>
<span class="co">#&gt;             5             3             2             5             5 </span>
<span class="co">#&gt;          EIF1      EIF4EBP1          ELOB          ENHO        ENTPD1 </span>
<span class="co">#&gt;             5             5             5             3             5 </span>
<span class="co">#&gt;          EREG         EVI2A         EVI2B       FAM120A       FAM198B </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;        FAM45A         FANCI          FCAR        FCER1A        FCER1G </span>
<span class="co">#&gt;             5             2             5             3             5 </span>
<span class="co">#&gt;        FCGR1A        FCGR2A         FCGRT          FCN1           FES </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;          FGD4          FGL2        FKBP1A         FKBP2          FLT3 </span>
<span class="co">#&gt;             5             5             5             3             3 </span>
<span class="co">#&gt;         FOLR3           FOS          FOSB         FOSL2          FPR1 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;          FTH1           FTL         FTLP3          G0S2     GABARAPL1 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;         GAPDH          GAS6          GAS7           GCA        GLIPR1 </span>
<span class="co">#&gt;             5             3             5             5             5 </span>
<span class="co">#&gt;          GLRX          GLUL          GMFG          GNAQ           GNS </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;        GPBAR1        GPCPD1          GPX1         GRINA           GRN </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;           GSN         GSTP1          GZMB         H2AFY         H3F3B </span>
<span class="co">#&gt;             3             5             3             5             5 </span>
<span class="co">#&gt;         HACD4           HCK      HIST1H4C           HK2      HLA-DPA1 </span>
<span class="co">#&gt;             5             5             2             5             3 </span>
<span class="co">#&gt;      HLA-DPB1      HLA-DQA1       HLA-DRA      HLA-DRB1      HLA-DRB6 </span>
<span class="co">#&gt;             3             3             3             3             3 </span>
<span class="co">#&gt;         HMGB2          HNMT         HSBP1      HSP90AB1     HSP90AB3P </span>
<span class="co">#&gt;             2             5             5             4             4 </span>
<span class="co">#&gt;        HSPA1A           ID1          IER2          IER3          IER5 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;         IFI30        IFITM3        IFNGR2         IGSF6       IL13RA1 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;          IL1B         IL1RN        IQGAP1         IRAK3       IRF2BP2 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;          IRF7          IRF8          IRS2         ITGAM         ITM2B </span>
<span class="co">#&gt;             3             3             5             5             5 </span>
<span class="co">#&gt;         ITM2C      IVNS1ABP          JAML           JUN          JUNB </span>
<span class="co">#&gt;             3             5             5             5             5 </span>
<span class="co">#&gt;          JUND         KCNE3        KCTD12      KIAA0930         KIF23 </span>
<span class="co">#&gt;             5             5             5             5             2 </span>
<span class="co">#&gt;         KLF10          KLF4          KLF6        KNSTRN         LAMP5 </span>
<span class="co">#&gt;             5             5             5             2             3 </span>
<span class="co">#&gt;       LAMTOR1       LAMTOR4          LCP1        LGALS1        LGALS2 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;        LGALS3          LGMN        LILRA2        LILRA4        LILRA5 </span>
<span class="co">#&gt;             5             3             5             3             5 </span>
<span class="co">#&gt;        LILRB2        LILRB3        LILRB4     LINC00877     LINC00996 </span>
<span class="co">#&gt;             5             5             3             5             3 </span>
<span class="co">#&gt;          LMO2          LRP1        LRRC26        LRRC36         LRRK2 </span>
<span class="co">#&gt;             5             5             3             3             5 </span>
<span class="co">#&gt;          LSM6          LST1         LTA4H          LY86          LY96 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;          LYST           LYZ          MAFB         MAP1A      MAPKAPK2 </span>
<span class="co">#&gt;             5             5             5             3             3 </span>
<span class="co">#&gt;        MARCKS          MCL1         MEGF9        METTL9         MFSD1 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;         MGST1         MGST2          MIDN          MNDA         MPEG1 </span>
<span class="co">#&gt;             5             3             5             5             5 </span>
<span class="co">#&gt;        MS4A6A         MS4A7         MSRB1         MYADM         MYBL2 </span>
<span class="co">#&gt;             5             5             5             5             3 </span>
<span class="co">#&gt;          MYL6          MZB1         NAMPT         NAPRT         NAPSB </span>
<span class="co">#&gt;             3             3             5             5             3 </span>
<span class="co">#&gt;          NCF2       NDUFA11        NDUFB1        NDUFS6         NFAM1 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;        NFKBIA        NFKBIZ         NINJ1         NLRP3        NOTCH2 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;          NPC2         NR4A2          NRG1        NT5DC2        NUDT16 </span>
<span class="co">#&gt;             5             5             5             3             5 </span>
<span class="co">#&gt;        NUP214        NUSAP1          OAZ1         ODF3B        OGFRL1 </span>
<span class="co">#&gt;             5             2             5             5             5 </span>
<span class="co">#&gt;        OSBPL8         OSCAR         OTUD1        P2RY13         PARK7 </span>
<span class="co">#&gt;             5             5             5             5             3 </span>
<span class="co">#&gt;         PCBP1          PCNA         PFDN5          PFN1           PGD </span>
<span class="co">#&gt;             5             1             5             2             5 </span>
<span class="co">#&gt;          PGLS          PHEX        PICALM          PID1         PILRA </span>
<span class="co">#&gt;             5             3             5             5             5 </span>
<span class="co">#&gt;        PLA2G7         PLAC8         PLAUR         PLBD1          PLD4 </span>
<span class="co">#&gt;             5             3             5             5             3 </span>
<span class="co">#&gt;          PLEK          PLP2        PLSCR1        PLXDC2        PLXNB2 </span>
<span class="co">#&gt;             3             3             5             5             5 </span>
<span class="co">#&gt;         PNRC1         POLE4        POLR2L          POMP      PPP1R14B </span>
<span class="co">#&gt;             5             5             5             5             3 </span>
<span class="co">#&gt;      PPP1R15A          PPT1         PRAM1         PRDX3       PRELID1 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;          PROC         PRR11          PSAP         PTAFR         PTGDS </span>
<span class="co">#&gt;             3             2             5             5             3 </span>
<span class="co">#&gt;         PTGS2         PTPRE         PTPRS         PTTG1        PYCARD </span>
<span class="co">#&gt;             5             5             3             2             5 </span>
<span class="co">#&gt;          PYGL           QKI          QPCT     RAB11FIP1         RAB31 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;         RAB32         RAB3D          RAC1       RACGAP1        RASSF2 </span>
<span class="co">#&gt;             5             5             5             2             5 </span>
<span class="co">#&gt;         RBM47          RBP7          RBX1          RETN          RGS2 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;          RHOA          RHOB          RHOG         RIPK2     RN7SL368P </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;     RNA5SP151        RNASE2        RNASE6        RNF130      RNU4ATAC </span>
<span class="co">#&gt;             5             5             3             5             5 </span>
<span class="co">#&gt;      RNVU1-19 RP11-1143G9.4  RP11-160E2.6  RP11-24F11.2 RP11-363K21.1 </span>
<span class="co">#&gt;             5             5             5             3             5 </span>
<span class="co">#&gt;   RP11-40C6.2 RP11-522I20.3  RP11-71G12.1  RP11-73G16.2  RP4-647C14.2 </span>
<span class="co">#&gt;             4             2             3             3             3 </span>
<span class="co">#&gt; RP5-1121H13.3        RPL13A        RPL23A        RPL27A          RPL3 </span>
<span class="co">#&gt;             2             4             4             4             4 </span>
<span class="co">#&gt;         RPL31          RPL4          RPL5         RPLP2         RPS12 </span>
<span class="co">#&gt;             4             4             4             4             4 </span>
<span class="co">#&gt;         RPS18          RPS2         RPS23         RPS24         RPS27 </span>
<span class="co">#&gt;             4             4             4             5             4 </span>
<span class="co">#&gt;        RPS27A         RPS29          RPS3          RPS5          RPS6 </span>
<span class="co">#&gt;             4             4             4             4             4 </span>
<span class="co">#&gt;          RPSA          RRM2          RTN1          RTN3          RTN4 </span>
<span class="co">#&gt;             4             2             5             5             5 </span>
<span class="co">#&gt;          RXRA       S100A10       S100A11       S100A12        S100A4 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;        S100A6        S100A8        S100A9        SAMHD1          SAT1 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;        SCAMP5         SCN9A           SCT         SDCBP        SEC11A </span>
<span class="co">#&gt;             3             3             3             5             5 </span>
<span class="co">#&gt;        SEC61B        SEC61G       SELENOS        SEMA4A         SERF2 </span>
<span class="co">#&gt;             3             3             3             5             5 </span>
<span class="co">#&gt;      SERPINA1      SERPINF1          SGK1       SH3BGRL      SH3BGRL3 </span>
<span class="co">#&gt;             5             3             5             5             3 </span>
<span class="co">#&gt;         SIRPA        SIRPB1       SLC11A1       SLC15A3       SLC16A3 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;       SLC31A2        SLC7A7        SLC8A1          SMC2         SMCO4 </span>
<span class="co">#&gt;             5             5             5             2             5 </span>
<span class="co">#&gt;        SMIM25         SMIM5         SMPD3     SNORD3B-2          SOD2 </span>
<span class="co">#&gt;             5             3             3             5             5 </span>
<span class="co">#&gt;         SORT1         SPCS1         SPDL1          SPI1          SRGN </span>
<span class="co">#&gt;             5             3             2             5             5 </span>
<span class="co">#&gt;         STAB1         STMN1         STX11        STXBP2         SULF2 </span>
<span class="co">#&gt;             5             2             5             5             3 </span>
<span class="co">#&gt;       SULT1A1        SYCP2L        TAGLN2        TALDO1          TCF4 </span>
<span class="co">#&gt;             5             3             3             5             3 </span>
<span class="co">#&gt;          TET2          TFEC         TGFBI        TIMM8B         TIMP1 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;         TIMP2           TKT          TLR2          TLR4          TLR8 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;       TMEM107      TMEM167A      TMEM170B        TMEM8B       TNFAIP2 </span>
<span class="co">#&gt;             5             5             5             3             5 </span>
<span class="co">#&gt;      TNFRSF1B      TNFRSF21       TNFSF10      TNFSF13B         TOP2A </span>
<span class="co">#&gt;             5             3             5             5             2 </span>
<span class="co">#&gt;          TOX2          TPI1          TPM2          TPP1          TPX2 </span>
<span class="co">#&gt;             2             5             3             5             2 </span>
<span class="co">#&gt;         TRAIP         TREM1         TRIB1          TSPO          TTI2 </span>
<span class="co">#&gt;             2             5             5             5             2 </span>
<span class="co">#&gt;        TUBA1A        TUBA1B          TUBB           TXN          TYMP </span>
<span class="co">#&gt;             5             2             2             3             5 </span>
<span class="co">#&gt;          TYMS        TYROBP         UBE2C          UGCG        UQCR11 </span>
<span class="co">#&gt;             2             5             2             3             5 </span>
<span class="co">#&gt;         VAMP8          VCAN           VIM          VSIR          WARS </span>
<span class="co">#&gt;             3             5             5             5             5 </span>
<span class="co">#&gt;          YBX3         YPEL5         YWHAE          ZEB2        ZFAND5 </span>
<span class="co">#&gt;             5             5             5             5             5 </span>
<span class="co">#&gt;         ZFP36       ZFP36L1       ZNF385A        ZNF467         ZWINT </span>
<span class="co">#&gt;             5             5             3             5             2</span>

<span class="fu"><a href="../reference/scdb_add_gset.html">scdb_add_gset</a></span><span class="op">(</span><span class="va">gset_nm</span>, <span class="va">gset</span><span class="op">)</span></code></pre></div>
<p>Note that we constructed a membership vector with genes as names and the number of the top correlated anchor gene as the value. This is optional, we could have created a gene set with all genes in the same set (e.g. 1). It is important to add the gene set to the database for the next steps. We use <em>gset_nm</em> as the gene set id.</p>
</div>
<div class="section level2">
<h2 id="fine-clustering-and-filtering-of-the-gene-set">Fine clustering and filtering of the gene set<a class="anchor" aria-label="anchor" href="#fine-clustering-and-filtering-of-the-gene-set"></a>
</h2>
<p>To further examine the gene set we got, we'll cluster the genes. First, we'll create a sub-matrix from the umi matrix that will contain only the genes in the gene set:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sub_mat_id</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"test"</span>, <span class="va">gset_nm</span>, sep<span class="op">=</span><span class="st">"_"</span><span class="op">)</span>
<span class="fu"><a href="../reference/mcell_mat_ignore_genes.html">mcell_mat_ignore_genes</a></span><span class="op">(</span>new_mat_id <span class="op">=</span> <span class="va">sub_mat_id</span>, mat_id <span class="op">=</span> <span class="st">"test"</span>, ig_genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">foc_genes</span><span class="op">)</span>, reverse <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
<p>Now we'll cluster the genes on the downsampled sub-matrix, using hierarchical clustering and cutting the tree to 20 clusters:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/mcell_gset_split_by_dsmat.html">mcell_gset_split_by_dsmat</a></span><span class="op">(</span>gset_id <span class="op">=</span> <span class="va">gset_nm</span>, mat_id <span class="op">=</span> <span class="va">sub_mat_id</span>, K <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>
<span class="co">#&gt; will downsample the matrix, N= 750</span></code></pre></div>
<p>The number of clusters should be proportional to the number of genes we cluster, as we want to get homogeneous clusters of coodinated genes.</p>
<p>The cluster membership information is updated in the gene set. Let's re-load it and have a look:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gset</span> <span class="op">=</span> <span class="fu"><a href="../reference/scdb_gset.html">scdb_gset</a></span><span class="op">(</span><span class="va">gset_nm</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">gset</span><span class="op">)</span>
<span class="co">#&gt; An object of class "tgGeneSets"</span>
<span class="co">#&gt; Slot "description":</span>
<span class="co">#&gt; [1] "Cell cycle and stress correlated genes hc K=20"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Slot "set_names":</span>
<span class="co">#&gt;  [1]  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Slot "gene_set":</span>
<span class="co">#&gt;    AC023590.1          ACTB         ACTG1        AGTRAP         AHNAK </span>
<span class="co">#&gt;             1             2             3             4             2 </span>
<span class="co">#&gt;           AHR          AIF1       ALDH1A1         ALDH2       ALDH3B1 </span>
<span class="co">#&gt;             5             6             7             8             7 </span>
<span class="co">#&gt;         ALDOA       ALOX5AP       ANKRD28         ANPEP         ANXA1 </span>
<span class="co">#&gt;             2             9             5             7             2 </span>
<span class="co">#&gt;         ANXA2         ANXA5          AOAH         AP1S2         AP2S1 </span>
<span class="co">#&gt;             2             2            10             4             6 </span>
<span class="co">#&gt;         APLP2           APP      ARHGEF40         ARL5B     ARL6IP1P2 </span>
<span class="co">#&gt;             4            11             7             5            12 </span>
<span class="co">#&gt;        ARRDC4         ASAH1         ASF1B         ASGR1          ASIP </span>
<span class="co">#&gt;             7            13            14             4            11 </span>
<span class="co">#&gt;          ATF3          ATG3         ATOX1    ATP2B1-AS1         ATP5E </span>
<span class="co">#&gt;            10            15             5            10             5 </span>
<span class="co">#&gt;        ATP5G2         ATP5I       ATP6V0B      ATP6V0D1      ATP6V0E1 </span>
<span class="co">#&gt;             5             5             6             7            16 </span>
<span class="co">#&gt;       ATP6V1A      ATP6V1B2       ATP6V1F         BACH1         BCAT1 </span>
<span class="co">#&gt;             7            15             7            15            12 </span>
<span class="co">#&gt;        BCL2A1         BIRC5       BLOC1S1         BLVRB        BNIP3L </span>
<span class="co">#&gt;            15            14             5             4             5 </span>
<span class="co">#&gt;          BRI3          BST1      C10orf11      C12orf75       C14orf2 </span>
<span class="co">#&gt;            15             4             7             9             7 </span>
<span class="co">#&gt;      C19orf38      C19orf47      C1orf162      C20orf27       C4orf48 </span>
<span class="co">#&gt;            13            12            15            15             7 </span>
<span class="co">#&gt;         C5AR1       C7orf73       C9orf72         CALM2         CAMK1 </span>
<span class="co">#&gt;            13             5            15            15            15 </span>
<span class="co">#&gt;          CAPG        CAPZA2        CARD16         CARS2         CASP1 </span>
<span class="co">#&gt;             8            15            15             7            15 </span>
<span class="co">#&gt;      CATSPER1       CCDC189        CCDC50       CCDC71L       CCDC88A </span>
<span class="co">#&gt;            10             9            11            12             7 </span>
<span class="co">#&gt;          CCL3        CCL3L3         CCNA2         CCNL1          CCNY </span>
<span class="co">#&gt;            10            10            14            17             5 </span>
<span class="co">#&gt;          CCR1          CCR9          CD14         CD163        CD300E </span>
<span class="co">#&gt;            15            12             4            10            15 </span>
<span class="co">#&gt;       CD300LF         CD302          CD33          CD36           CD4 </span>
<span class="co">#&gt;            15            10             4             4            15 </span>
<span class="co">#&gt;          CD44          CD63          CD70          CD83          CD86 </span>
<span class="co">#&gt;            16             8            12            15            15 </span>
<span class="co">#&gt;          CD93           CDA         CDC42         CDCA8          CDK1 </span>
<span class="co">#&gt;            10            13            16            14            14 </span>
<span class="co">#&gt;          CDK2        CDKN2C         CDKN3          CDR2         CEBPB </span>
<span class="co">#&gt;            12            12            14            16            13 </span>
<span class="co">#&gt;         CEBPD         CECR1         CENPE         CENPF         CENPN </span>
<span class="co">#&gt;             6            10            14            14            12 </span>
<span class="co">#&gt;         CENPP         CENPU           CFD           CFP        CHMP1B </span>
<span class="co">#&gt;            14            14             6             6            16 </span>
<span class="co">#&gt;          CHP1          CIB2         CKAP4          CKLF       CLEC10A </span>
<span class="co">#&gt;             7             9             7             2            18 </span>
<span class="co">#&gt;       CLEC12A        CLEC4A        CLEC4C        CLEC4E        CLEC7A </span>
<span class="co">#&gt;            13             4            11             4             6 </span>
<span class="co">#&gt;         CMTM6         CNPY3          COMT        CORO1A         COTL1 </span>
<span class="co">#&gt;            15             8             5             3             2 </span>
<span class="co">#&gt;        COX6B1         COX7B        CPPED1          CPVL         CREB5 </span>
<span class="co">#&gt;             5             5            15             6             7 </span>
<span class="co">#&gt;         CREG1         CRTAP          CRYM         CSF3R          CST3 </span>
<span class="co">#&gt;             7             7             1             4             6 </span>
<span class="co">#&gt;          CSTA    CTB-61M7.2  CTC-510F12.4  CTD-3252C9.4          CTSB </span>
<span class="co">#&gt;             6             7             7            10             8 </span>
<span class="co">#&gt;          CTSC          CTSD          CTSH          CTSS          CTSZ </span>
<span class="co">#&gt;             9             4            19             6             6 </span>
<span class="co">#&gt;         CXCL2         CXCL8          CYBA          CYBB        CYP1B1 </span>
<span class="co">#&gt;             7            10             6             6             7 </span>
<span class="co">#&gt;        DAZAP2         DERL3         DMXL2      DNASE1L3          DOK3 </span>
<span class="co">#&gt;             5            11            15            11            15 </span>
<span class="co">#&gt;          DPM2          DPYD        DPYSL2        DSTNP2         DUSP1 </span>
<span class="co">#&gt;            12             5            10            12            17 </span>
<span class="co">#&gt;         DUSP6        DYNLL1       EFCAB11          EGR1       EHBP1L1 </span>
<span class="co">#&gt;            13             5            12            17             7 </span>
<span class="co">#&gt;          EIF1      EIF4EBP1          ELOB          ENHO        ENTPD1 </span>
<span class="co">#&gt;            17            15             5            18            15 </span>
<span class="co">#&gt;          EREG         EVI2A         EVI2B       FAM120A       FAM198B </span>
<span class="co">#&gt;             7            16            15             5             7 </span>
<span class="co">#&gt;        FAM45A         FANCI          FCAR        FCER1A        FCER1G </span>
<span class="co">#&gt;             7            12             7            18             6 </span>
<span class="co">#&gt;        FCGR1A        FCGR2A         FCGRT          FCN1           FES </span>
<span class="co">#&gt;             7            15             6             6             8 </span>
<span class="co">#&gt;          FGD4          FGL2        FKBP1A         FKBP2          FLT3 </span>
<span class="co">#&gt;            15             6            15            12             9 </span>
<span class="co">#&gt;         FOLR3           FOS          FOSB         FOSL2          FPR1 </span>
<span class="co">#&gt;             5            17            17             7             4 </span>
<span class="co">#&gt;          FTH1           FTL         FTLP3          G0S2     GABARAPL1 </span>
<span class="co">#&gt;             6             6            13             7             7 </span>
<span class="co">#&gt;         GAPDH          GAS6          GAS7           GCA        GLIPR1 </span>
<span class="co">#&gt;             2             1             5             4             8 </span>
<span class="co">#&gt;          GLRX          GLUL          GMFG          GNAQ           GNS </span>
<span class="co">#&gt;             7             5            16             5            15 </span>
<span class="co">#&gt;        GPBAR1        GPCPD1          GPX1         GRINA           GRN </span>
<span class="co">#&gt;            13             7             4            15             4 </span>
<span class="co">#&gt;           GSN         GSTP1          GZMB         H2AFY         H3F3B </span>
<span class="co">#&gt;            18             6            11             8            17 </span>
<span class="co">#&gt;         HACD4           HCK      HIST1H4C           HK2      HLA-DPA1 </span>
<span class="co">#&gt;            15            13             3             7            19 </span>
<span class="co">#&gt;      HLA-DPB1      HLA-DQA1       HLA-DRA      HLA-DRB1      HLA-DRB6 </span>
<span class="co">#&gt;            19            19            19            19            19 </span>
<span class="co">#&gt;         HMGB2          HNMT         HSBP1      HSP90AB1     HSP90AB3P </span>
<span class="co">#&gt;            17            10             6            20            12 </span>
<span class="co">#&gt;        HSPA1A           ID1          IER2          IER3          IER5 </span>
<span class="co">#&gt;             5            10            17            15             7 </span>
<span class="co">#&gt;         IFI30        IFITM3        IFNGR2         IGSF6       IL13RA1 </span>
<span class="co">#&gt;            13            13            15             6            15 </span>
<span class="co">#&gt;          IL1B         IL1RN        IQGAP1         IRAK3       IRF2BP2 </span>
<span class="co">#&gt;            10             7            15             7             5 </span>
<span class="co">#&gt;          IRF7          IRF8          IRS2         ITGAM         ITM2B </span>
<span class="co">#&gt;            11            11            16             7            16 </span>
<span class="co">#&gt;         ITM2C      IVNS1ABP          JAML           JUN          JUNB </span>
<span class="co">#&gt;            11            16             8            17            17 </span>
<span class="co">#&gt;          JUND         KCNE3        KCTD12      KIAA0930         KIF23 </span>
<span class="co">#&gt;            10             7            10            15            12 </span>
<span class="co">#&gt;         KLF10          KLF4          KLF6        KNSTRN         LAMP5 </span>
<span class="co">#&gt;            10            10            17            12             1 </span>
<span class="co">#&gt;       LAMTOR1       LAMTOR4          LCP1        LGALS1        LGALS2 </span>
<span class="co">#&gt;            15             5             5             6             6 </span>
<span class="co">#&gt;        LGALS3          LGMN        LILRA2        LILRA4        LILRA5 </span>
<span class="co">#&gt;             6             9            15            11            13 </span>
<span class="co">#&gt;        LILRB2        LILRB3        LILRB4     LINC00877     LINC00996 </span>
<span class="co">#&gt;            13            13             1            15            11 </span>
<span class="co">#&gt;          LMO2          LRP1        LRRC26        LRRC36         LRRK2 </span>
<span class="co">#&gt;            15            15             1            12            15 </span>
<span class="co">#&gt;          LSM6          LST1         LTA4H          LY86          LY96 </span>
<span class="co">#&gt;             5             6            15            19            10 </span>
<span class="co">#&gt;          LYST           LYZ          MAFB         MAP1A      MAPKAPK2 </span>
<span class="co">#&gt;            13             6            13             1             1 </span>
<span class="co">#&gt;        MARCKS          MCL1         MEGF9        METTL9         MFSD1 </span>
<span class="co">#&gt;            15             5             7             7             7 </span>
<span class="co">#&gt;         MGST1         MGST2          MIDN          MNDA         MPEG1 </span>
<span class="co">#&gt;             4             8             7             6             6 </span>
<span class="co">#&gt;        MS4A6A         MS4A7         MSRB1         MYADM         MYBL2 </span>
<span class="co">#&gt;             4            13             7            16             1 </span>
<span class="co">#&gt;          MYL6          MZB1         NAMPT         NAPRT         NAPSB </span>
<span class="co">#&gt;             2            11            13             5            11 </span>
<span class="co">#&gt;          NCF2       NDUFA11        NDUFB1        NDUFS6         NFAM1 </span>
<span class="co">#&gt;            13             5             5             5            15 </span>
<span class="co">#&gt;        NFKBIA        NFKBIZ         NINJ1         NLRP3        NOTCH2 </span>
<span class="co">#&gt;            10            15            15             7            15 </span>
<span class="co">#&gt;          NPC2         NR4A2          NRG1        NT5DC2        NUDT16 </span>
<span class="co">#&gt;             6            17             7            12            15 </span>
<span class="co">#&gt;        NUP214        NUSAP1          OAZ1         ODF3B        OGFRL1 </span>
<span class="co">#&gt;             4            14             6            13            10 </span>
<span class="co">#&gt;        OSBPL8         OSCAR         OTUD1        P2RY13         PARK7 </span>
<span class="co">#&gt;            12            15             7             5             3 </span>
<span class="co">#&gt;         PCBP1          PCNA         PFDN5          PFN1           PGD </span>
<span class="co">#&gt;             5            12            16             2             7 </span>
<span class="co">#&gt;          PGLS          PHEX        PICALM          PID1         PILRA </span>
<span class="co">#&gt;            15             1             7            10            13 </span>
<span class="co">#&gt;        PLA2G7         PLAC8         PLAUR         PLBD1          PLD4 </span>
<span class="co">#&gt;             7             3            13             4            11 </span>
<span class="co">#&gt;          PLEK          PLP2        PLSCR1        PLXDC2        PLXNB2 </span>
<span class="co">#&gt;             8             9            15            10            15 </span>
<span class="co">#&gt;         PNRC1         POLE4        POLR2L          POMP      PPP1R14B </span>
<span class="co">#&gt;            17             5             7             5            11 </span>
<span class="co">#&gt;      PPP1R15A          PPT1         PRAM1         PRDX3       PRELID1 </span>
<span class="co">#&gt;            17            15            13            15            15 </span>
<span class="co">#&gt;          PROC         PRR11          PSAP         PTAFR         PTGDS </span>
<span class="co">#&gt;            11            12             6             7             1 </span>
<span class="co">#&gt;         PTGS2         PTPRE         PTPRS         PTTG1        PYCARD </span>
<span class="co">#&gt;             7            10             9            14             6 </span>
<span class="co">#&gt;          PYGL           QKI          QPCT     RAB11FIP1         RAB31 </span>
<span class="co">#&gt;             7             7             4             7             7 </span>
<span class="co">#&gt;         RAB32         RAB3D          RAC1       RACGAP1        RASSF2 </span>
<span class="co">#&gt;             4             7             6            12             7 </span>
<span class="co">#&gt;         RBM47          RBP7          RBX1          RETN          RGS2 </span>
<span class="co">#&gt;             7             4             5             7             6 </span>
<span class="co">#&gt;          RHOA          RHOB          RHOG         RIPK2     RN7SL368P </span>
<span class="co">#&gt;             5            13             2             7             7 </span>
<span class="co">#&gt;     RNA5SP151        RNASE2        RNASE6        RNF130      RNU4ATAC </span>
<span class="co">#&gt;             7             7             8             6             7 </span>
<span class="co">#&gt;      RNVU1-19 RP11-1143G9.4  RP11-160E2.6  RP11-24F11.2 RP11-363K21.1 </span>
<span class="co">#&gt;            16             4             7             9            15 </span>
<span class="co">#&gt;   RP11-40C6.2 RP11-522I20.3  RP11-71G12.1  RP11-73G16.2  RP4-647C14.2 </span>
<span class="co">#&gt;            20            12             1            11             1 </span>
<span class="co">#&gt; RP5-1121H13.3        RPL13A        RPL23A        RPL27A          RPL3 </span>
<span class="co">#&gt;            16            20            20            20            20 </span>
<span class="co">#&gt;         RPL31          RPL4          RPL5         RPLP2         RPS12 </span>
<span class="co">#&gt;            20            20            20            20            20 </span>
<span class="co">#&gt;         RPS18          RPS2         RPS23         RPS24         RPS27 </span>
<span class="co">#&gt;            20            20            20            12            20 </span>
<span class="co">#&gt;        RPS27A         RPS29          RPS3          RPS5          RPS6 </span>
<span class="co">#&gt;            20            20            20            20            20 </span>
<span class="co">#&gt;          RPSA          RRM2          RTN1          RTN3          RTN4 </span>
<span class="co">#&gt;            20            14            15             5             5 </span>
<span class="co">#&gt;          RXRA       S100A10       S100A11       S100A12        S100A4 </span>
<span class="co">#&gt;            15             2             6             4             6 </span>
<span class="co">#&gt;        S100A6        S100A8        S100A9        SAMHD1          SAT1 </span>
<span class="co">#&gt;             6             4             4             5             6 </span>
<span class="co">#&gt;        SCAMP5         SCN9A           SCT         SDCBP        SEC11A </span>
<span class="co">#&gt;             1             9            11            15             5 </span>
<span class="co">#&gt;        SEC61B        SEC61G       SELENOS        SEMA4A         SERF2 </span>
<span class="co">#&gt;             9             9             9            15             5 </span>
<span class="co">#&gt;      SERPINA1      SERPINF1          SGK1       SH3BGRL      SH3BGRL3 </span>
<span class="co">#&gt;             6            11            10            15             2 </span>
<span class="co">#&gt;         SIRPA        SIRPB1       SLC11A1       SLC15A3       SLC16A3 </span>
<span class="co">#&gt;            15             7            13            15             7 </span>
<span class="co">#&gt;       SLC31A2        SLC7A7        SLC8A1          SMC2         SMCO4 </span>
<span class="co">#&gt;            15            13            15            12            15 </span>
<span class="co">#&gt;        SMIM25         SMIM5         SMPD3     SNORD3B-2          SOD2 </span>
<span class="co">#&gt;            13            11             1            15             7 </span>
<span class="co">#&gt;         SORT1         SPCS1         SPDL1          SPI1          SRGN </span>
<span class="co">#&gt;            15             9            12             6             6 </span>
<span class="co">#&gt;         STAB1         STMN1         STX11        STXBP2         SULF2 </span>
<span class="co">#&gt;             7             3            15             6             7 </span>
<span class="co">#&gt;       SULT1A1        SYCP2L        TAGLN2        TALDO1          TCF4 </span>
<span class="co">#&gt;            15            12             9             8            11 </span>
<span class="co">#&gt;          TET2          TFEC         TGFBI        TIMM8B         TIMP1 </span>
<span class="co">#&gt;             5             7             8             5             2 </span>
<span class="co">#&gt;         TIMP2           TKT          TLR2          TLR4          TLR8 </span>
<span class="co">#&gt;             7            13            15             7             7 </span>
<span class="co">#&gt;       TMEM107      TMEM167A      TMEM170B        TMEM8B       TNFAIP2 </span>
<span class="co">#&gt;             5             7             7             9            10 </span>
<span class="co">#&gt;      TNFRSF1B      TNFRSF21       TNFSF10      TNFSF13B         TOP2A </span>
<span class="co">#&gt;            15            11            15             4            14 </span>
<span class="co">#&gt;          TOX2          TPI1          TPM2          TPP1          TPX2 </span>
<span class="co">#&gt;             1             2            11             8            14 </span>
<span class="co">#&gt;         TRAIP         TREM1         TRIB1          TSPO          TTI2 </span>
<span class="co">#&gt;            12             7             7             6            12 </span>
<span class="co">#&gt;        TUBA1A        TUBA1B          TUBB           TXN          TYMP </span>
<span class="co">#&gt;            17             3             3             9             6 </span>
<span class="co">#&gt;          TYMS        TYROBP         UBE2C          UGCG        UQCR11 </span>
<span class="co">#&gt;            14             6            14            11            16 </span>
<span class="co">#&gt;         VAMP8          VCAN           VIM          VSIR          WARS </span>
<span class="co">#&gt;             5             4             2             2            13 </span>
<span class="co">#&gt;          YBX3         YPEL5         YWHAE          ZEB2        ZFAND5 </span>
<span class="co">#&gt;            10            17             8             6            15 </span>
<span class="co">#&gt;         ZFP36       ZFP36L1       ZNF385A        ZNF467         ZWINT </span>
<span class="co">#&gt;            17             5            15             7            14</span></code></pre></div>
<p>However, it is much more convenient to look at the correlation heatmaps of these clusters, so let's generate those:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/mcell_plot_gset_cor_mats.html">mcell_plot_gset_cor_mats</a></span><span class="op">(</span>gset_id <span class="op">=</span> <span class="va">gset_nm</span>, scmat_id <span class="op">=</span> <span class="va">sub_mat_id</span><span class="op">)</span>
<span class="co">#&gt; will downsample the matrix, N= 750</span></code></pre></div>
<p>The plots are generated under a subdirectory of the figures directory named <em>gset_nm</em>.gset_cors. Common practice is to examine them and select the clusters that we want to keep. In our example, In this case we would probably keep clusters 3, 5, 10, 14, 16 and 17 in the gene set: <img src="figs/lateral.gset_cors/3.png" alt="clus3"><img src="figs/lateral.gset_cors/5.png" alt="clus5"><img src="figs/lateral.gset_cors/10.png" alt="clus10"><img src="figs/lateral.gset_cors/14.png" alt="clus14"><img src="figs/lateral.gset_cors/16.png" alt="clus16"><img src="figs/lateral.gset_cors/17.png" alt="clus17"></p>
<p>The rest of the clusters contain mostly genes that are correlated to cell size but not related to the cell cycle.</p>
<p>To create the final filtered gene set, we'll keep only the clusters we've selected in the gene set, and we'll save the filtered gene set in the database:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/mcell_gset_remove_clusts.html">mcell_gset_remove_clusts</a></span><span class="op">(</span>gset_id <span class="op">=</span> <span class="va">gset_nm</span>, filt_clusts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">5</span>, <span class="fl">10</span>, <span class="fl">14</span>, <span class="fl">16</span>, <span class="fl">17</span><span class="op">)</span>, new_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">gset_nm</span>, <span class="st">"_filtered"</span><span class="op">)</span>, reverse<span class="op">=</span><span class="cn">T</span><span class="op">)</span>

<span class="va">lateral_gset_id</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">gset_nm</span>, <span class="st">"_filtered"</span><span class="op">)</span>
<span class="va">lateral_gset</span> <span class="op">=</span> <span class="fu"><a href="../reference/scdb_gset.html">scdb_gset</a></span><span class="op">(</span><span class="va">lateral_gset_id</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">lateral_gset</span><span class="op">)</span>
<span class="co">#&gt; An object of class "tgGeneSets"</span>
<span class="co">#&gt; Slot "description":</span>
<span class="co">#&gt; [1] "Cell cycle and stress correlated genes hc K=20 cl_filt"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Slot "set_names":</span>
<span class="co">#&gt; [1]  3  5 10 14 16 17</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Slot "gene_set":</span>
<span class="co">#&gt;         ACTG1           AHR       ANKRD28          AOAH         ARL5B </span>
<span class="co">#&gt;             3             5             5            10             5 </span>
<span class="co">#&gt;         ASF1B          ATF3         ATOX1    ATP2B1-AS1         ATP5E </span>
<span class="co">#&gt;            14            10             5            10             5 </span>
<span class="co">#&gt;        ATP5G2         ATP5I      ATP6V0E1         BIRC5       BLOC1S1 </span>
<span class="co">#&gt;             5             5            16            14             5 </span>
<span class="co">#&gt;        BNIP3L       C7orf73      CATSPER1          CCL3        CCL3L3 </span>
<span class="co">#&gt;             5             5            10            10            10 </span>
<span class="co">#&gt;         CCNA2         CCNL1          CCNY         CD163         CD302 </span>
<span class="co">#&gt;            14            17             5            10            10 </span>
<span class="co">#&gt;          CD44          CD93         CDC42         CDCA8          CDK1 </span>
<span class="co">#&gt;            16            10            16            14            14 </span>
<span class="co">#&gt;         CDKN3          CDR2         CECR1         CENPE         CENPF </span>
<span class="co">#&gt;            14            16            10            14            14 </span>
<span class="co">#&gt;         CENPP         CENPU        CHMP1B          COMT        CORO1A </span>
<span class="co">#&gt;            14            14            16             5             3 </span>
<span class="co">#&gt;        COX6B1         COX7B  CTD-3252C9.4         CXCL8        DAZAP2 </span>
<span class="co">#&gt;             5             5            10            10             5 </span>
<span class="co">#&gt;          DPYD        DPYSL2         DUSP1        DYNLL1          EGR1 </span>
<span class="co">#&gt;             5            10            17             5            17 </span>
<span class="co">#&gt;          EIF1          ELOB         EVI2A       FAM120A         FOLR3 </span>
<span class="co">#&gt;            17             5            16             5             5 </span>
<span class="co">#&gt;           FOS          FOSB          GAS7          GLUL          GMFG </span>
<span class="co">#&gt;            17            17             5             5            16 </span>
<span class="co">#&gt;          GNAQ         H3F3B      HIST1H4C         HMGB2          HNMT </span>
<span class="co">#&gt;             5            17             3            17            10 </span>
<span class="co">#&gt;        HSPA1A           ID1          IER2          IL1B       IRF2BP2 </span>
<span class="co">#&gt;             5            10            17            10             5 </span>
<span class="co">#&gt;          IRS2         ITM2B      IVNS1ABP           JUN          JUNB </span>
<span class="co">#&gt;            16            16            16            17            17 </span>
<span class="co">#&gt;          JUND        KCTD12         KLF10          KLF4          KLF6 </span>
<span class="co">#&gt;            10            10            10            10            17 </span>
<span class="co">#&gt;       LAMTOR4          LCP1          LSM6          LY96          MCL1 </span>
<span class="co">#&gt;             5             5             5            10             5 </span>
<span class="co">#&gt;         MYADM         NAPRT       NDUFA11        NDUFB1        NDUFS6 </span>
<span class="co">#&gt;            16             5             5             5             5 </span>
<span class="co">#&gt;        NFKBIA         NR4A2        NUSAP1        OGFRL1        P2RY13 </span>
<span class="co">#&gt;            10            17            14            10             5 </span>
<span class="co">#&gt;         PARK7         PCBP1         PFDN5          PID1         PLAC8 </span>
<span class="co">#&gt;             3             5            16            10             3 </span>
<span class="co">#&gt;        PLXDC2         PNRC1         POLE4          POMP      PPP1R15A </span>
<span class="co">#&gt;            10            17             5             5            17 </span>
<span class="co">#&gt;         PTPRE         PTTG1          RBX1          RHOA      RNVU1-19 </span>
<span class="co">#&gt;            10            14             5             5            16 </span>
<span class="co">#&gt; RP5-1121H13.3          RRM2          RTN3          RTN4        SAMHD1 </span>
<span class="co">#&gt;            16            14             5             5             5 </span>
<span class="co">#&gt;        SEC11A         SERF2          SGK1         STMN1          TET2 </span>
<span class="co">#&gt;             5             5            10             3             5 </span>
<span class="co">#&gt;        TIMM8B       TMEM107       TNFAIP2         TOP2A          TPX2 </span>
<span class="co">#&gt;             5             5            10            14            14 </span>
<span class="co">#&gt;        TUBA1A        TUBA1B          TUBB          TYMS         UBE2C </span>
<span class="co">#&gt;            17             3             3            14            14 </span>
<span class="co">#&gt;        UQCR11         VAMP8          YBX3         YPEL5         ZFP36 </span>
<span class="co">#&gt;            16             5            10            17            17 </span>
<span class="co">#&gt;       ZFP36L1         ZWINT </span>
<span class="co">#&gt;             5            14</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="generating-metacells-while-blacklisting-our-lateral-gene-set">Generating metacells while blacklisting our lateral gene set<a class="anchor" aria-label="anchor" href="#generating-metacells-while-blacklisting-our-lateral-gene-set"></a>
</h2>
<p>This section will follow the exact same steps performed in the <strong>Running metacell analysis: guided tutorial on 8K PBMCs</strong> vignette in order to generate the K-nn graph, the co-clustering matrix and the metacells, with the small change of disabling the lateral genes from affecting the process.</p>
<p>We start by removing the lateral genes from the list of feature genes:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">marker_gset</span> <span class="op">=</span> <span class="fu"><a href="../reference/scdb_gset.html">scdb_gset</a></span><span class="op">(</span><span class="st">"test_feats"</span><span class="op">)</span>   
<span class="va">marker_gset</span> <span class="op">=</span> <span class="fu"><a href="../reference/gset_new_restrict_gset.html">gset_new_restrict_gset</a></span><span class="op">(</span>gset <span class="op">=</span> <span class="va">marker_gset</span>, filt_gset <span class="op">=</span> <span class="va">lateral_gset</span>, inverse <span class="op">=</span> <span class="cn">T</span>, desc <span class="op">=</span> <span class="st">"cgraph gene markers w/o lateral genes"</span><span class="op">)</span>
<span class="fu"><a href="../reference/scdb_add_gset.html">scdb_add_gset</a></span><span class="op">(</span><span class="st">"test_feats_filtered"</span>, <span class="va">marker_gset</span><span class="op">)</span></code></pre></div>
<p>We'll now generate the graph using the filtered list of feature genes:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/mcell_add_cgraph_from_mat_bknn.html">mcell_add_cgraph_from_mat_bknn</a></span><span class="op">(</span>mat_id<span class="op">=</span><span class="st">"test"</span>, 
                gset_id <span class="op">=</span> <span class="st">"test_feats_filtered"</span>, 
                graph_id<span class="op">=</span><span class="st">"test_graph_filtered"</span>,
                K<span class="op">=</span><span class="fl">100</span>,
                dsamp<span class="op">=</span><span class="cn">T</span><span class="op">)</span>
<span class="co">#&gt; will downsample the matrix, N= 1877</span>
<span class="co">#&gt; will build balanced knn graph on 8276 cells and 851 genes, this can be a bit heavy for &gt;20,000 cells</span>
<span class="co">#&gt; sim graph is missing 36 nodes, out of 8276</span></code></pre></div>
<p>And now simply following the steps of the <strong>Running metacell analysis: guided tutorial on 8K PBMCs</strong> vignette from the K-nn graph until the colored metacell object:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/mcell_coclust_from_graph_resamp.html">mcell_coclust_from_graph_resamp</a></span><span class="op">(</span>
                coc_id<span class="op">=</span><span class="st">"test_coc500_filtered"</span>, 
                graph_id<span class="op">=</span><span class="st">"test_graph_filtered"</span>,
                min_mc_size<span class="op">=</span><span class="fl">20</span>, 
                p_resamp<span class="op">=</span><span class="fl">0.75</span>, n_resamp<span class="op">=</span><span class="fl">500</span><span class="op">)</span>
<span class="co">#&gt; running bootstrap to generate cocluster</span>
<span class="co">#&gt; done resampling</span>

<span class="fu"><a href="../reference/mcell_mc_from_coclust_balanced.html">mcell_mc_from_coclust_balanced</a></span><span class="op">(</span>
                coc_id<span class="op">=</span><span class="st">"test_coc500_filtered"</span>, 
                mat_id<span class="op">=</span> <span class="st">"test"</span>,
                mc_id<span class="op">=</span> <span class="st">"test_filtered_mc"</span>, 
                K<span class="op">=</span><span class="fl">30</span>, min_mc_size<span class="op">=</span><span class="fl">30</span>, alpha<span class="op">=</span><span class="fl">2</span><span class="op">)</span>
<span class="co">#&gt; filtered 5027554 left with 742806 based on co-cluster imbalance</span>
<span class="co">#&gt; building metacell object, #mc 75</span>
<span class="co">#&gt; add batch counts</span>
<span class="co">#&gt; compute footprints</span>
<span class="co">#&gt; compute absolute ps</span>
<span class="co">#&gt; compute coverage ps</span>
<span class="co">#&gt; reordering metacells by hclust and most variable two markers</span>
<span class="co">#&gt; reorder on CD74 vs LEF1</span>

<span class="fu"><a href="../reference/mcell_mc_split_filt.html">mcell_mc_split_filt</a></span><span class="op">(</span>new_mc_id<span class="op">=</span><span class="st">"test_filtered_mc_f"</span>, 
            mc_id<span class="op">=</span><span class="st">"test_filtered_mc"</span>, 
            mat_id<span class="op">=</span><span class="st">"test"</span>,
            T_lfc<span class="op">=</span><span class="fl">3</span>, plot_mats<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="co">#&gt; starting split outliers</span>
<span class="co">#&gt; add batch counts</span>
<span class="co">#&gt; compute footprints</span>
<span class="co">#&gt; compute absolute ps</span>
<span class="co">#&gt; compute coverage ps</span>

<span class="va">marks_colors</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"pbmc_mc_colorize.txt"</span>, package<span class="op">=</span><span class="st">"metacell"</span><span class="op">)</span>, sep<span class="op">=</span><span class="st">"\t"</span>, h<span class="op">=</span><span class="cn">T</span>, stringsAsFactors<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="fu"><a href="../reference/mc_colorize.html">mc_colorize</a></span><span class="op">(</span><span class="st">"test_filtered_mc_f"</span>, marker_colors<span class="op">=</span><span class="va">marks_colors</span><span class="op">)</span></code></pre></div>
<p>It is convenient not to restrict the selection of genes that will be displayed in the heatmap of genes and metacells, but to mark the lateral genes that did not affect the partitioning of cells to metacells:</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/mcell_gset_from_mc_markers.html">mcell_gset_from_mc_markers</a></span><span class="op">(</span>gset_id<span class="op">=</span><span class="st">"test_filtered_markers"</span>, mc_id<span class="op">=</span><span class="st">"test_filtered_mc_f"</span>, blacklist_gset_id<span class="op">=</span><span class="va">lateral_gset_id</span><span class="op">)</span>
<span class="fu"><a href="../reference/mcell_gset_from_mc_markers.html">mcell_gset_from_mc_markers</a></span><span class="op">(</span>gset_id<span class="op">=</span><span class="st">"test_filtered_markers_lat"</span>, mc_id<span class="op">=</span><span class="st">"test_filtered_mc_f"</span>, filt_gset_id<span class="op">=</span><span class="va">lateral_gset_id</span><span class="op">)</span>
    
<span class="fu"><a href="../reference/mcell_mc_plot_marks.html">mcell_mc_plot_marks</a></span><span class="op">(</span>mc_id<span class="op">=</span><span class="st">"test_filtered_mc_f"</span>, gset_id<span class="op">=</span><span class="st">"test_filtered_markers"</span>, mat_id<span class="op">=</span><span class="st">"test"</span>, lateral_gset<span class="op">=</span><span class="st">"test_filtered_markers_lat"</span><span class="op">)</span>
<span class="co">#&gt; setting fig h to 1000 md levels 0 num of marks 47</span></code></pre></div>
<p><img src="figs/test_filtered_mc_f.cells_heat_marks.png" alt="heatmap_marks"> The marker genes in the <em>test_filtered_markers</em> gene set do not contain any lateral gene (shown in black in the heatmap), and the <em>test_filtered_markers_lat</em> gene set contain the lateral genes that were selected as markers (shown in red).</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="http://compgenomics.weizmann.ac.il/tanay/" class="external-link">Amos Tanay</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
